docs/proxmox.md

```diff
@@ -0,0 +1,94 @@
+# Proxmox Virtual Environment
+
+## State of the implementation
+
+Support for Proxmox as a provider in the machine-controller is currently just a technical demo. It
+is possible to create MachineDeployments using manually created VM templates as demonstrated below.
+In this example the VM template is using local storage, which is why this template can only be
+cloned on the same node it is located at.
+
+## Prerequisites
+
+### Authentication
+
+For authentication the following data is needed:
+
+- `user_id` is expected to be in the form `USER@REALM!TOKENID`
+- `token` is just the UUID you get when initially creating the token
+
+See also:
+* https://pve.proxmox.com/wiki/User_Management#pveum_tokens
+* https://pve.proxmox.com/wiki/Proxmox_VE_API#API_Tokens
+
+#### User Privileges
+
+For the provider to properly function the user needs an API token with the following privileges:
+
+* `Datastore.AllocateSpace`
+* `Pool.Allocate`
+* `Pool.Audit`
+* `VM.Allocate`
+* `VM.Audit`
+* `VM.Clone`
+* `VM.Config.CDROM`
+* `VM.Config.CPU`
+* `VM.Config.Cloudinit`
+* `VM.Config.Disk`
+* `VM.Config.HWType`
+* `VM.Config.Memory`
+* `VM.Config.Network`
+* `VM.Config.Options`
+* `VM.Monitor`
+* `Sys.Audit`
+* `Sys.Console`
+
+### Cloud-Init enabled VM Templates
```

Comment:
nit: I think it is supposed to be just "Cloud-init" when capitalized, otherwise "cloud-init" per the official docs:
https://cloudinit.readthedocs.io/_/downloads/en/20.2/pdf/

> There are some common words that should follow specific usage: 
> * cloud-init: always lower case with a hyphen, unless starting a sentence in which case only the ‘C’ is capitalized (e.g. Cloud-init).